src/Microsoft.AspNetCore.Sockets/WebSockets.cs

```diff
@@ -3,143 +3,163 @@
 
 using System;
 using System.Net.WebSockets;
-using System.Threading;
 using System.Threading.Tasks;
 using Channels;
 using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebSockets.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.WebSockets.Internal;
+using WebSocketCloseStatus = Microsoft.Extensions.WebSockets.Internal.WebSocketCloseStatus;
 
 namespace Microsoft.AspNetCore.Sockets
 {
     public class WebSockets : IHttpTransport
     {
+#if NET451
+        private static readonly Task CompletedTask = Task.FromResult(0);
+#else
+        private static readonly Task CompletedTask = Task.CompletedTask;
+#endif
+
+        private static readonly TimeSpan _closeTimeout = TimeSpan.FromSeconds(5);
+        private static readonly WebSocketAcceptContext EmptyContext = new WebSocketAcceptContext();
+
         private readonly HttpChannel _channel;
-        private readonly Connection _connection;
-        private readonly WebSocketMessageType _messageType;
+        private readonly WebSocketOpcode _opcode;
+        private readonly ILogger _logger;
 
-        public WebSockets(Connection connection, Format format)
+        public WebSockets(Connection connection, Format format, ILoggerFactory loggerFactory)
         {
-            _connection = connection;
-            _channel = (HttpChannel)connection.Channel;
-            _messageType = format == Format.Binary ? WebSocketMessageType.Binary : WebSocketMessageType.Text;
+            _channel = (HttpChannel) connection.Channel;
+            _opcode = format == Format.Binary ? WebSocketOpcode.Binary : WebSocketOpcode.Text;
+
+            _logger = (ILogger) loggerFactory?.CreateLogger<WebSockets>() ?? NullLogger.Instance;
         }
 
         public async Task ProcessRequestAsync(HttpContext context)
         {
-            if (!context.WebSockets.IsWebSocketRequest)
+            var feature = context.Features.Get<IHttpWebSocketConnectionFeature>();
+            if (feature == null || !feature.IsWebSocketRequest)
```

Comment:
It has to be `feature?.IsWebSocketRequest ?? false`, which just seems more verbose to me.